eng/generators.targets

```diff
@@ -63,8 +63,34 @@
                             and @(ProjectReference->AnyHaveMetadataValue('Identity', '$(CoreLibProject)'))))" Include="$(LibrariesProjectRoot)Common\src\System\Runtime\InteropServices\ArrayMarshaller.cs" />
   </ItemGroup>
 
+  <ItemGroup>
+    <EnabledGenerators Include="RegexGenerator" Condition="'$(EnableRegexGenerator)' == 'true'" />
+    <!-- Enable the RegexGenerator source generator when the project is a C# source project that references System.Text.RegularExpressions -->
+    <EnabledGenerators Include="RegexGenerator"
+                       Condition="'$(EnableRegexGenerator)' == ''
+                        and '$(IsFrameworkSupportFacade)' != 'true'
+                        and '$(IsSourceProject)' == 'true'
+                        and '$(MSBuildProjectExtension)' == '.csproj'
+                        and ('@(Reference)' != '' and @(Reference->AnyHaveMetadataValue('Identity', 'System.Text.RegularExpressions')))" />
+    <EnabledGenerators Include="RegexGenerator"
+                       Condition="'$(EnableRegexGenerator)' == ''
+                        and '$(IsFrameworkSupportFacade)' != 'true'
+                        and '$(IsSourceProject)' == 'true'
+                        and '$(MSBuildProjectExtension)' == '.csproj'
+                        and ('$(TargetFrameworkIdentifier)' == '.NETStandard' or '$(TargetFrameworkIdentifier)' == '.NETFramework')" />
```

Comment:
@eerhardt and I had some offline discussions about making source generators available to frameworks where they actually can't really be consumed outside of dotnet/runtime (which is the case for the Regex one) and I believe we landed on the idea that we shouldn't do this, and instead either: a) limit the source generator to only be used on platforms where it can be consumed externally, or b) expose it to other frameworks if we would see value for adding them.

If you agree, can we limit this change to only add the source generator in the netcoreapp case? (basically remove changes from line 75-80 here)

Reply:
I'm fine with this, but what effect is this going to have in other areas, like tests?  If a project has `net48` in `TargetFrameworks`, would that mean it couldn't be used there?  What about tests that have `PlatformDetection.IsNetFramework`?

Reply:
> If a project has net48 in TargetFrameworks, would that mean it couldn't be used there?

Correct. That behavior aligns with our customers' experience. The same goes for `netstandard2.0` and `net6.0`. Only `net7.0`+ targets should be able to consume the RegexGenerator.

Reply:
Right, so any project that is expected to reference .NetF _at all_ can't use the generator unless we get fancy with `#ifdef`s, then, which is something I'll have to consider for some of the other test projects.

Reply:
As an aside,  while technically at the moment the generated code would likely work downlevel, changes are coming that will end up using new API surface area in .NET 7, so the generator will only be usable with .NET 7+.